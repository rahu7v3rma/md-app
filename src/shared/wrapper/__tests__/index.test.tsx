import React from 'react';
import {
    ActivityIndicator,
    RefreshControl,
    ScrollView,
    Text
} from 'react-native';
import { act, create, ReactTestRenderer } from 'react-test-renderer';

import Wrapper from '@/shared/wrapper';

const mockLoading = false;
const mockChildren = <Text>mockChildren</Text>;
const mockOnRefresh = jest.fn().mockImplementation(() => {});
const mockOnScroll = jest.fn().mockImplementation((_) => {});

let tree: ReactTestRenderer;
beforeAll(() => {
    tree = create(
        <Wrapper
            loading={mockLoading}
            onRefresh={mockOnRefresh}
            onScroll={mockOnScroll}
        >
            {mockChildren}
        </Wrapper>
    );
});

test('Snapshot Testing with Jest', () => {
    expect(tree.toJSON()).toMatchSnapshot();
});

test('renders without crashing', () => {
    expect(tree.root).toBeTruthy();
});

test('does not display loading indicator when loading is false', () => {
    expect(tree.root.findAllByType(ActivityIndicator)).toHaveLength(0);
});

test('calls onRefresh callback when RefreshControl is triggered', async () => {
    await act(() => {
        tree.root.findByType(RefreshControl).props.onRefresh();
    });
    expect(mockOnRefresh).toBeCalled();
});

test('calls onScroll callback when ScrollView is scrolled', async () => {
    await act(() => {
        tree.root.findByType(ScrollView).props.onScrollEndDrag();
    });
    expect(mockOnScroll).toBeCalled();
});
