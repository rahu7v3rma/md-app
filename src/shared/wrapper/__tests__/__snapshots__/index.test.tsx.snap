// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing with Jest 1`] = `
<RCTScrollView
  bounces={true}
  onScrollEndDrag={[MockFunction]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[MockFunction]}
      refreshing={false}
    />
  }
  scrollEnabled={true}
  style={
    {
      "height": "100%",
    }
  }
>
  <RCTRefreshControl />
  <View>
    <Text>
      mockChildren
    </Text>
  </View>
</RCTScrollView>
`;
